service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/history/SealedOrderHistoryService.java

```diff
@@ -109,6 +117,39 @@ public GeneratedOrder lastGeneratedOrder(CaseData caseData) {
             .getValue();
     }
 
+    public Map<String, Object> processUploadedOrder(CaseData caseData) {
+        Map<String, Object> data = new HashMap<>();
+        // get the latest order just created in about-to-submit callback;
+        List<Element<GeneratedOrder>> pastOrders = caseData.getOrderCollection();
+        GeneratedOrder order = caseData.getOrderCollection().get(0).getValue();
+
+        DocumentReference sealedPDForder = orderCreationService.createOrderDocument(caseData, OrderStatus.SEALED, PDF);
+        DocumentReference plainWordOrder = orderCreationService.createOrderDocument(caseData, OrderStatus.PLAIN, WORD);
+
+        GeneratedOrder sealedOrder = GeneratedOrder.builder()
+            .orderType(order.getOrderType()) // hidden field, to store the type
+            .title(order.getTitle())
+            .type(order.getType())
+            .markedFinal(order.getMarkedFinal())
+            .children(order.getChildren())
+            .others(order.getOthers()) // hidden field, to store the selected others for notify
+            .judgeAndLegalAdvisor(order.getJudgeAndLegalAdvisor())
+            .dateTimeIssued(order.getDateTimeIssued())
+            .approvalDate(order.getApprovalDate())
+            .approvalDateTime(order.getApprovalDateTime())
+            .childrenDescription(order.getChildrenDescription())
+            .specialGuardians(order.getSpecialGuardians())
+            .othersNotified(order.getOthersNotified())
+            .document(sealedPDForder)
+            .translationRequirements(order.getTranslationRequirements())
+            .unsealedDocumentCopy(plainWordOrder)
+            .build();
```

Comment:
might be able to just do order.toBuilder() to save copying over all the attributes and only update the document/sealedDoc attributes

Reply:
Oh yeah, thx for the advice. I will change it.